core/runtime.rs

```diff
@@ -486,13 +504,17 @@ impl JsRuntime {
     }
 
     let state = state_rc.borrow();
-    let is_idle = {
-      state.pending_ops.is_empty()
-        && state.pending_dyn_imports.is_empty()
-        && state.preparing_dyn_imports.is_empty()
+    let is_idle =
+      { state.pending_ops.is_empty() && state.pending_dyn_imports.is_empty() };
+
+    let has_pending_dyn_imports = !{
+      state.preparing_dyn_imports.is_empty()
+        && state.pending_dyn_mod_evaluate.is_empty()
     };
 
-    if is_idle {
+    let has_pending_module = state.pending_mod_evaluate.is_some();
+
+    if is_idle && !has_pending_dyn_imports && !has_pending_module {
```

Comment:
Maybe add ` !has_pending_dyn_imports && !has_pending_module` to the `is_idle` calculation? This bit of code is confusing and I don't understand why it's broken up into multiple statements.

Reply:
Refactored and added comment